### ISSUE TITLE
Failure of the Audio Output device to change spontaneously when anyone is connected to the video conference room.

### ISSUE THREAD

**Issue by devosonder** [livekit/client-sdk-js#967]
### Describe the bug

@livekit/components-react@^1.5.0
@livekit/components-core@^0.8.2
livekit-client@^1.15.4
OS:Windows 11
Node JS: v20.9.0
Windows default sound output device: Speakers
Windows other audio output device: Headphones

A person joining a video conference room changes the Audio Output setting from Speakers to Headphones.
All audio is coming from the headphones. While the video conference is in progress, another person joins the conference. The Audio output setting of all users in the video conference room automatically changes to Windows default sound output and the sound starts coming from the speakers.

https://github.com/livekit/client-sdk-js/tree/main/example also has this bug.

### Reproduction

User1 and User2 join the room.
User1's windows audio output device is the "speakers". After entering the room, he selects the Headset as the audio output device from the https://github.com/livekit/client-sdk-js/tree/main/example web interface.
User1 hears User2's voice through the headset.
User2 disconnects and reconnects.
User1 now hears the sounds through the speaker.

### System Info

```shell
wsl2 latest version
Clients Browser: Microsoft Edge Version 120.0.2210.77 (Official build) (64-bit)
System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (32) x64 AMD Ryzen 9 7950X 16-Core Processor
    Memory: 14.49 GB / 15.62 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v20.9.0/bin/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    pnpm: 8.12.1 - ~/.nvm/versions/node/v20.9.0/bin/pnpm
  npmPackages:
    livekit-client: ^1.15.4 => 1.15.4
```

### Severity

annoyance

**Comment by davidzhao** [comment]
Which browser is the receiver using?

**Comment by devosonder** [comment]
> Which browser is the receiver using?

All chromium based browsers have this problem. I tested it using Microsoft Edge Version 120.0.2210.77 (Official build) (64-bit).

**Comment by devosonder** [comment]
```
livekitRoom.on(RoomEvent.TrackSubscribed, () => {
      const activeAudioOutputDevice = livekitRoom.getActiveAudioOutputDevice();
      livekitRoom.switchActiveDevice('audiooutput', activeAudioOutputDevice, true);
 });
```
If I set the deviceId I got with livekitRoom.getActiveAudioOutputDevice() in RoomEvent.TrackSubscribed event with livekitRoom.switchActiveDevice('audiooutput', activeAudioOutputDevice, true);, it works properly. I don't think TrackSubscribed event is a suitable event to solve this problem. 

The getActiveAudioOutputDevice() method gives the correct deviceId, but the sound comes out from a different device (windows default output device).

By writing the above code I only found a temporary solution, the BUG still exists.

Translated with www.DeepL.com/Translator (free version)
